src/entity/relation.rs

```diff
@@ -97,9 +155,19 @@ impl RelationDef {
             is_owner: !self.is_owner,
             on_delete: self.on_delete,
             on_update: self.on_update,
+            on_condition: self.on_condition,
             fk_name: None,
         }
     }
+
+    /// Set custom join ON condition
+    pub fn on_condition<F>(mut self, f: F) -> Self
+    where
+        F: Fn(DynIden, DynIden) -> Condition + 'static,
```

Comment:
May be typedef this signature and add some documents?

Reply:
Ah trait alias is still nightly.

Reply:
But we need to at least document what is fn is supposed to do and what the parameters mean